api/envoy/extensions/wasm/v3/wasm.proto

```diff
@@ -50,7 +49,7 @@ message VmConfig {
   // [#comment: TODO: add ref for details.]
   string vm_id = 1;
 
-  // The Wasm runtime type.
+  // The Wasm runtime type, defaults to wasm engine used at Envoy build-time.
```

Comment:
Technically it is possible to build multiple engines in Envoy (e.g. wasmtime and V8 doesn't conflict), what will be default then?

Reply:
Can you define multiple `wasm` values? Defining `wasm` twice results in only the latter value taking effect:
```
$ bazel build --config=libc++ --define wasm=v8 --define wasm=wasmtime //source/exe:envoy-static
```
```
$ bazel-bin/source/exe/envoy-static 2>&1 | grep wasm.runtime
[...]   envoy.wasm.runtime: envoy.wasm.runtime.null, envoy.wasm.runtime.wasmtime
```

Reply:
Current bazel configuration doesn't allow this, but it doesn't mean any future build or a downstream build cannot build multiple wasm engines in same binaries as they are registered as extensions. So we still need to define a behavior for that case.

Reply:
Right, but the documentation can be updated once this capability becomes available. I find it a bit misleading to say that "the first configured Wasm engine will be used" when we support only one.

Reply:
My point here is the Envoy API isn't fully bound to Envoy implementation, there's other implementations and there are downstream builds of Envoy as well. There is no deterministic way to find "the first configured Wasm engine will be used" when we build more than one engines. If we want to make a default engine selection here, either hardcode one or hardcode a priority list from ones that is supported today, so there's no ambiguity?

Reply:
@bryanwux could you make the change requested by @lizan?

Reply:
I am not sure if I understand correctly. What @lizan says about "hardcode one" here means we still need to define the wasm runtime in `vm_config.runtime` since bazel may support multiple wasm engines in one envoy build. So we should not make `vm_config.runtime` here?

Reply:
No, he means that you could hardcode a list and use first one that's available, so that the order is deterministic, e.g.:

- Check if `envoy.wasm.runtime.v8` is available. If it is, then use it.
- Otherwise, check if `envoy.wasm.runtime.wasmtime` is available. If it is, then use it.
- Otherwise, check if `envoy.wasm.runtime.wamr` is available. If it is, then use it.
- Otherwise, check if `envoy.wasm.runtime.wavm` is available. If it is, then use it.

This doesn't make a difference (vs querying which engine was used at the build time), since there is only one available in Envoy anyway (at least, right now).